src/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -614,10 +614,13 @@ private static void parseSingleAltAllele(List<Allele> alleles, String alt, int l
     }
 
     public final static boolean canDecodeFile(final String potentialInput, final String MAGIC_HEADER_LINE) {
-        try {
-            return isVCFStream(new FileInputStream(potentialInput), MAGIC_HEADER_LINE) ||
-                    isVCFStream(new GZIPInputStream(new FileInputStream(potentialInput)), MAGIC_HEADER_LINE) ||
-                    isVCFStream(new BlockCompressedInputStream(new FileInputStream(potentialInput)), MAGIC_HEADER_LINE);
+        try(
+            final FileInputStream fis = new FileInputStream(potentialInput);
```

Comment:
i can't imagine `canDecodeFile` being a bottleneck and rewriting this may make it less readable. Let's leave as is for now, wdyt?